kubeflow/e2e_example.py

```diff
@@ -10,90 +10,136 @@
-    num_gpus = 1
+    (add_env(generate_dataset_op, repo_versions)
+        .add_node_selector_constraint('beta.kubernetes.io/instance-type', 'm5.4xlarge')
+    )
+   
+    train_repos = repo_versions.copy()
+    train_repos.pop('WORKDIR_REPO')
+    train_repos.pop('WORKDIR_VERSION')
+    train_num_gpus=1
     train_op = components.load_component_from_file('components/train.yaml')(
             image=image,
             s3_bucket=s3_bucket,
             owner=owner,
             dataset_owner=dataset_owner,
-            task_name='almond_dialogue_nlu',
+            task_name=train_task_name,
             project=project,
             experiment=experiment,
             dataset=dataset,
             model=model,
-            load_from='None',
-            additional_args=train_args) 
+            load_from=train_load_from,
+            s3_datadir=generate_dataset_op.outputs['s3_datadir'],
+            train_iterations=train_iterations,
+            save_every=train_save_every,
+            log_every=train_log_every,
+            val_every=train_val_every,
+            additional_args=train_additional_args) 
     (train_op.container
         .set_memory_request('56Gi')
         .set_memory_limit('56Gi')
         .set_cpu_request('7.5')
         .set_cpu_limit('7.5')
-        .set_gpu_limit(str(num_gpus))
+        .set_gpu_limit(str(train_num_gpus))
         .add_volume_mount(V1VolumeMount(name='tensorboard', mount_path='/shared/tensorboard'))
     )
-    (disable_caching(train_op)
+    (add_env(train_op, train_repos)
         .add_toleration(V1Toleration(key='nvidia.com/gpu', operator='Exists', effect='NoSchedule'))
-        .add_node_selector_constraint('beta.kubernetes.io/instance-type', f'p3.{2*num_gpus}xlarge')
+        .add_node_selector_constraint('beta.kubernetes.io/instance-type', f'p3.{2*train_num_gpus}xlarge')
         .add_volume(V1Volume(name='tensorboard',
             persistent_volume_claim=V1PersistentVolumeClaimVolumeSource('tensorboard-research-kf')))
         .after(generate_dataset_op)
     )
     
     eval_op = components.load_component_from_file('components/evaluate.yaml')(
             image=image,
-            s3_bucket=s3_bucket,
-            owner=owner,
             project=project,
             experiment=experiment,
             model=model,
             model_owner=owner,
-            eval_set='dev',
-            eval_version='eval_version',
-            additional_args='')  
+            eval_set=eval_set,
+            eval_version=eval_version,
+            s3_model_dir=train_op.outputs['s3_model_dir'],
+            additional_args=eval_additional_args)  
     (eval_op.container
         .set_memory_limit('15Gi')
         .set_memory_request('15Gi')
         .set_cpu_limit('4')
         .set_cpu_request('4'))  
-    (disable_caching(eval_op)
+    (add_env(eval_op, repo_versions)
         .after(train_op)
     )
+
+if __name__ == '__main__':
+    resp = upload_pipeline('genie_train', train_pipeline)
```

Comment:
Shall we call this just "train"? Or perhaps "generate-train-eval"? We now have "train" and "genie_train" and it's already confusing. Soon, we'll also have a pipeline that goes generate -> train -> paraphrase -> train -> eval.

Reply:
`generate-train-eval` sounds good